engine/classes/ElggWidgetDefinition.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * \ElggWidgetDefinition
+ *
+ * Helper class for defining a widget
+ *
+ * @package    Elgg.Core
+ * @subpackage Widgets
+ *
+ */
+class ElggWidgetDefinition {
+
+	/**
+	 * @var string Handler of the widget
+	 */
+	public $handler;
+	
+	/**
+	 * \ElggWidgetDefinition constructor
+	 *
+	 * @param string $handler Handler of the widget
```

Comment:
I'd be for changing as much core stuff to $id as we can, but I don't know what data structures leak out to plugins. __get for "handler" would be OK, but if this is a new class, why is it needed?